python/tests/integration/arcticdb/test_arctic_batch.py

```diff
@@ -0,0 +1,962 @@
+    for sym in range(num_symbols):
+        original_dataframe = list_dataframes[sym]
+        assert read_batch_result[sym].metadata == "great_metadata" + str(num_versions - 1)
+        assert_frame_equal(read_batch_result[sym].data, original_dataframe)
+
+    num_segments = int(
+        (num_days * num_rows_per_day) / lib._nvs._lib_cfg.lib_desc.version.write_options.segment_row_size
+    )
+    for sym in range(num_symbols):
+        data_key_version = lib._nvs.read_index("symbol_" + str(sym))["version_id"]
+        for s in range(num_segments):
+            assert data_key_version[s] == 0
+
+
+def test_read_batch_time_stamp(arctic_library):
+    """Should be able to read data in batch mode using a timestamp."""
+    lib = arctic_library
+    sym = "sym_"
+    num_versions = 3
+    num_symbols = 3
+    for v_num in range(num_versions):
+        write_requests = []
+        for sym_num in range(num_symbols):
+            write_requests.append(
+                WritePayload(
+                    sym + str(sym_num), pd.DataFrame({"col": [sym_num + v_num, sym_num * v_num, sym_num - v_num]})
+                )
+            )
+
+    microsecond_delta = timedelta(microseconds=1)
+
+    data_non_batch = [
+        lib.read(
+            sym + str(sym_num),
+            as_of=datetime(
+                (version_info.date + microsecond_delta).year,
+                (version_info.date + microsecond_delta).month,
+                (version_info.date + microsecond_delta).day,
+                (version_info.date + microsecond_delta).hour,
+                (version_info.date + microsecond_delta).minute,
+                (version_info.date + microsecond_delta).second,
+                tzinfo=timezone.utc,
+            ),
+        )
+        for sym_num in range(num_symbols)
+        for version_info in lib.list_versions(sym + str(sym_num))
+    ]
+
+    requests_batch = [
+        ReadRequest(
+            sym + str(sym_num),
+            as_of=datetime(
+                (version_info.date + microsecond_delta).year,
+                (version_info.date + microsecond_delta).month,
+                (version_info.date + microsecond_delta).day,
+                (version_info.date + microsecond_delta).hour,
+                (version_info.date + microsecond_delta).minute,
+                (version_info.date + microsecond_delta).second,
+                tzinfo=timezone.utc,
+            ),
+        )
+        for sym_num in range(num_symbols)
+        for version_info in lib.list_versions(sym + str(sym_num))
+    ]
+
+    original_dataframes = [
+        pd.DataFrame(
+            {"col": [sym_num + version_info.version, sym_num * version_info.version, sym_num - version_info.version]}
+        )
+        for sym_num in range(num_symbols)
+        for version_info in lib.list_versions(sym + str(sym_num))
+    ]
+
+    data_batch = lib.read_batch(requests_batch)
+    for d1, d2, d3 in zip(data_non_batch, data_batch, original_dataframes):
```

Comment:
What's the point comparing against `data_non_batch`?

Reply:
it was just to re-assure the non-batch/batch behaviours were the same. But fine to removed this extra check